File: zb1/codingTest/220323/x/2.ReverseSentence/ReverseSentence.py
import re


def solution(s):
    """
    :param s: str
    :return: str[]
    """

    tokens = re.split(r'[.,!? ]', s)
    removedEmptyTokens = filter(lambda word: word != '', tokens)
    reversedTokens = map(lambda word: word[::-1], removedEmptyTokens)
    return list(reversedTokens)

File: zb1/codingTest/220323/x/2.ReverseSentence/ReverseSentence.js
/**
 * @param s {string}
 * @return {string[]}
 */
function solution(s) {
  return s
    .split(/[.,!? ]/) //정규식 사용
    .filter((word) => word !== "") //""가 아닌 요소를 모아 새로운 배열로 반환
    .map((word) => word.split("").reverse().join("")); //ex) abc->["a","b","c"]->["c","b","a"]->cba
}

solution;
